benchmark/napi/function_args/napi_binding.c

```diff
@@ -0,0 +1,227 @@
+    assert(status == napi_ok);
+
+    status = napi_get_named_property(env, args[0], "operand", &value);
+    assert(status == napi_ok);
+
+    status = napi_get_named_property(env, args[0], "data", &value);
+    assert(status == napi_ok);
+
+    status = napi_get_named_property(env, args[0], "reduce", &value);
+    assert(status == napi_ok);
+  }
+
+  return NULL;
+}
+
+static napi_value CallWithTypedarray(napi_env env, napi_callback_info info) {
+  napi_status status;
+
+  size_t argc = 1;
+  napi_value args[1];
+  status = napi_get_cb_info(env, info, &argc, args, NULL, NULL);
+  assert(status == napi_ok);
+
+  bool is_typedarray = false;
+  status = napi_is_typedarray(env, args[0], &is_typedarray);
+  assert(status == napi_ok);
+
+  assert(is_typedarray);
+  if (is_typedarray) {
+    napi_typedarray_type type;
+    napi_value input_buffer;
+    size_t byte_offset = 0;
+    size_t length = 0;
+    status = napi_get_typedarray_info(env, args[0], &type, &length,
+        NULL, &input_buffer, &byte_offset);
+    assert(status == napi_ok);
+    assert(length > 0);
+
+    void* data = NULL;
+    size_t byte_length = 0;
+    status = napi_get_arraybuffer_info(env,
+        input_buffer, &data, &byte_length);
+    assert(status == napi_ok);
+
+    uint32_t* input_integers = (uint32_t*)((uint8_t*)(data) + byte_offset);
+    assert(input_integers);
+  }
+
+  return NULL;
+}
+
+static napi_value CallWithArguments(napi_env env, napi_callback_info info) {
+  napi_status status;
+
+  size_t argc = 1;
+  napi_value args[1000];
+  // Get the length
+  status = napi_get_cb_info(env, info, &argc, NULL, NULL, NULL);
+  assert(status == napi_ok);
+
+  status = napi_get_cb_info(env, info, &argc, args, NULL, NULL);
+  assert(status == napi_ok);
+  assert(argc <= 1000);
+
+  napi_valuetype types[1];
+  status = napi_typeof(env, args[0], types);
+  assert(status == napi_ok);
+
+  assert(argc > 1 && types[0] == napi_number);
+  if (argc > 1 && types[0] == napi_number) {
+    uint32_t loop = 0;
+    status = napi_get_value_uint32(env, args[0], &loop);
+    assert(status == napi_ok);
+
+    for (uint32_t i = 1; i < loop; ++i) {
+      assert(i < argc);
+      status = napi_typeof(env, args[i], types);
+      assert(status == napi_ok);
+      assert(types[0] == napi_number);
+
+      uint32_t value = 0;
+      status = napi_get_value_uint32(env, args[i], &value);
+      assert(status == napi_ok);
+    }
+  }
+
+  return NULL;
+}
+
+
+#define EXPORT_FUNC(name, func) \
+  do { \
+    napi_value func ## _v; \
+    status = napi_create_function(env, \
+                                  name, \
+                                  NAPI_AUTO_LENGTH, \
+                                  func, \
+                                  NULL, \
+                                  &func ## _v); \
+    assert(status == napi_ok); \
+    status = napi_set_named_property(env, exports, name, func ## _v); \
+    assert(status == napi_ok); \
+  } while (0);
```

Comment:
nit: The trailing backslashes must be aligned to the column of the outermost trailing backslash.